Sources/ChatManager.swift

```diff
@@ -170,16 +180,30 @@ import NotificationCenter
                     return
                 }
 
+                var existingRooms: [PCRoom] = []
+
                 // If the currentUser property is already set then the assumption is that there was
                 // already a user subscription and so instead of setting the property to a new
                 // PCCurrentUser, we update the existing one to have the most up-to-date state
                 if let currentUser = strongSelf.currentUser {
+                    // We need to take copies of the rooms so that when we make the comparisons
+                    // of received rooms to pre-existing rooms we aren't actually just comparing
+                    // the same rooms (as PCRoom is a class and we have reference semantics)
+                    existingRooms = currentUser.rooms.map { $0.copy() }
```

Comment:
Could probably remove the need to take a copy by tracking the differences while you're updating the store. Not sure how much of a reshuffle this would necessitate. Maybe not worth it.

Reply:
Don't think it's worth it for now